src/hooks/useSendMessage.ts

```diff
@@ -64,8 +90,31 @@ function useSendMessage({ isRunning, setIsRunning, setExecutionStatus, message,
         options.signer = injector.signer;
         sourceAccount = account.address;
       }
-      await bridgeMessage.signAndSend(sourceAccount, { ...options });
-      setExecutionStatus(message.successfull);
+
+      const unsub = await bridgeMessage.signAndSend(sourceAccount, { ...options }, ({ events = [], status }) => {
+        if (status.isInBlock) {
+          events.forEach(({ event: { data, method } }) => {
+            if (method.toString() === 'MessageAccepted') {
+              const messageNonce = data.toArray()[1].toString();
+              sourceApi.rpc.chain.getBlock(status.asInBlock).then((res) => {
+                const block = res.block.header.number.toNumber();
```

Comment:
Why does it have to be a number here? In theory block numbers can be `u128` which is way more than what JS number  can store. I'd prefer to defer any numeric conversions as long as possible (i.e. only convert if it's strictly required for comparison or arithmetic) - this will make it easier to introduce `BigNumber` in the future if needed.